core/Db/Adapter.php

```diff
@@ -52,7 +52,9 @@ public static function factory($adapterName, & $dbInfos, $connect = true)
             try {
                 $adapter->getConnection();
 
-                Zend_Db_Table::setDefaultAdapter($adapter);
+                if (!Zend_Db_Table::getDefaultAdapter()) {
+                    Zend_Db_Table::setDefaultAdapter($adapter);
```

Comment:
In theory this actually changes the behavior. If the factory is called twice, with different adapters, the default adapter wouldn't be update on the second call anymore. But assuming we actually should always use the same adapter, guess this might be ok.

Reply:
I've removed this change as `Zend_Db_Table::setDefaultAdapter(null);` works well